### ISSUE TITLE
Cannot connect to Frida 15.0.x

### ISSUE THREAD

**Issue by cryptax** [MobSF/Mobile-Security-Framework-MobSF#1793]
## ENVIRONMENT

```
OS and Version: Linux Mint 20.2
Python Version: 3.8.10
MobSF Version: 3.4.5 Beta
Frida: 15.0.16
```

## EXPLANATION OF THE ISSUE

I cannot use MobSF Dynamic analysis any longer: it is unable to connect to Frida. 

```
[ERROR] 23/Aug/2021 13:34:08 - Error Connecting to Frida
Traceback (most recent call last):
  File "/home/axelle/softs/Mobile-Security-Framework-MobSF/mobsf/DynamicAnalyzer/views/android/frida_core.py", line 129, in connect
    pid = device.spawn([self.package])
  File "/home/axelle/softs/Mobile-Security-Framework-MobSF/venv/lib/python3.8/site-packages/frida/core.py", line 26, in wrapper
    return f(*args, **kwargs)
  File "/home/axelle/softs/Mobile-Security-Framework-MobSF/venv/lib/python3.8/site-packages/frida/core.py", line 149, in spawn
    return self._impl.spawn(program, argv, envp, env, cwd, stdio, aux_options)
frida.NotSupportedError: need Gadget to attach on jailed Android; its default location is: /home/axelle/.cache/frida/gadget-android-arm64.so
```

Everything used to work fine a month or two ago. MobSF would automatically download Frida and MobSFy my Android emulator. Now, it does not work. Perhaps: it used to work with Frida 14.x, now with Frida 15.x, it does not.

If I try Frida, though, it looks like it is working:

```
~/softs/Mobile-Security-Framework-MobSF$ frida-ps -U
  PID  Name
-----  ---------------------------------------------------
 8392  Calendar                                           
 8440  Clock                                              
 3923  Contacts                                           
 7289  Drive                                              
 7170  Files                                              
 2304  Google App                                         
 8472  Wallpapers                                         
 6200  adbd                                               
 1433  android.hardware.audio@2.0-service                 
 1553  android.hardware.biometrics.fingerprint@2.1-service
 1435  android.hardware.camera.provider@2.4-service       
 1436  android.hardware.configstore@1.0-service           
```

- I do not have `/home/axelle/.cache/frida/gadget-android-arm64.so` on my host.
- The emulator is seen fine:

```
$ adb devices
List of devices attached
emulator-5554	device
```

## STEPS TO REPRODUCE THE ISSUE

1. Install MobSF
2. Launch Android emulator 8 `~/Android/Sdk/emulator/emulator -avd Android8 -writable-system &> /dev/null &`
3. Load an APK, and select Dynamic analysis
4. Start Instrumentation. You get the error.

## LOG FILE

```
[INFO] 23/Aug/2021 13:33:28 - 
  __  __       _    ____  _____         _____ _  _   
 |  \/  | ___ | |__/ ___||  ___| __   _|___ /| || |  
 | |\/| |/ _ \| '_ \___ \| |_    \ \ / / |_ \| || |_ 
 | |  | | (_) | |_) |__) |  _|    \ V / ___) |__   _|
 |_|  |_|\___/|_.__/____/|_|       \_/ |____(_) |_|  

[INFO] 23/Aug/2021 13:33:28 - Mobile Security Framework v3.4.5 Beta
[INFO] 23/Aug/2021 13:33:28 - OS: Linux
[INFO] 23/Aug/2021 13:33:28 - Platform: Linux-5.10.0-1044-oem-x86_64-with-glibc2.29
[INFO] 23/Aug/2021 13:33:29 - Dist: linuxmint 20.2 uma
[INFO] 23/Aug/2021 13:33:29 - MobSF Basic Environment Check
[INFO] 23/Aug/2021 13:33:29 - Checking for Update.
[INFO] 23/Aug/2021 13:33:29 - No updates available.
[INFO] 23/Aug/2021 13:33:45 - Creating Dynamic Analysis Environment for com.egov.app
[INFO] 23/Aug/2021 13:33:45 - ADB Restarted
[INFO] 23/Aug/2021 13:33:45 - Waiting for 2 seconds...
[INFO] 23/Aug/2021 13:33:47 - Connecting to Android emulator-5554
[INFO] 23/Aug/2021 13:33:47 - Waiting for 2 seconds...
[INFO] 23/Aug/2021 13:33:49 - Restarting ADB Daemon as root
[INFO] 23/Aug/2021 13:33:49 - Waiting for 2 seconds...
[INFO] 23/Aug/2021 13:33:51 - Reconnecting to Android Device
[INFO] 23/Aug/2021 13:33:51 - Waiting for 2 seconds...
[INFO] 23/Aug/2021 13:33:54 - Found Android Studio Emulator
[INFO] 23/Aug/2021 13:33:54 - Remounting
[INFO] 23/Aug/2021 13:33:54 - Performing System check
[INFO] 23/Aug/2021 13:33:54 - Android API Level identified as 26
[INFO] 23/Aug/2021 13:33:54 - Android Version identified as 8.0
[INFO] 23/Aug/2021 13:33:54 - Environment MobSFyed Check
[INFO] 23/Aug/2021 13:33:54 - Installing MobSF RootCA
[INFO] 23/Aug/2021 13:33:54 - Starting HTTPs Proxy on 1337
[INFO] 23/Aug/2021 13:33:54 - Enabling ADB Reverse TCP on 1337
[INFO] 23/Aug/2021 13:33:54 - Setting Global Proxy for Android VM
[INFO] 23/Aug/2021 13:33:54 - Starting Clipboard Monitor
[INFO] 23/Aug/2021 13:33:54 - Getting screen resolution
[INFO] 23/Aug/2021 13:33:54 - Removing existing installation
[INFO] 23/Aug/2021 13:33:54 - Installing APK - com.egov.app
[INFO] 23/Aug/2021 13:33:54 - Testing Environment is Ready!
[INFO] 23/Aug/2021 13:34:06 - Starting Instrumentation
[INFO] 23/Aug/2021 13:34:06 - Starting Frida Server
[INFO] 23/Aug/2021 13:34:06 - Waiting for 2 seconds...
[ERROR] 23/Aug/2021 13:34:08 - Error Connecting to Frida
Traceback (most recent call last):
  File "/home/axelle/softs/Mobile-Security-Framework-MobSF/mobsf/DynamicAnalyzer/views/android/frida_core.py", line 129, in connect
    pid = device.spawn([self.package])
  File "/home/axelle/softs/Mobile-Security-Framework-MobSF/venv/lib/python3.8/site-packages/frida/core.py", line 26, in wrapper
    return f(*args, **kwargs)
  File "/home/axelle/softs/Mobile-Security-Framework-MobSF/venv/lib/python3.8/site-packages/frida/core.py", line 149, in spawn
    return self._impl.spawn(program, argv, envp, env, cwd, stdio, aux_options)
frida.NotSupportedError: need Gadget to attach on jailed Android; its default location is: /home/axelle/.cache/frida/gadget-android-arm64.so
```

**Comment by cryptax** [comment]
Ok, I managed to get it to work. It seems the problem was that I installed MobSF on a host which already had Frida. This somehow conflicted with the Frida for MobSF.
So, I erased my `~/.MobSF` directory, erased Frida from my host (if I need it, I'll put it in a virtual environment, that's always a better idea), and started setup again. This time, dynamic analysis works.

I'll close this issue, but there is perhaps something fishy with MobSF's Frida installation. Worth to check.
